.github/workflows/preprod-deployment.yml

```diff
@@ -83,20 +85,41 @@ jobs:
           SHORT_SHA: ${{ steps.vars.outputs.sha }}
 
       - name: Upload the VCS Agent binary to downloads.spacelift.dev
+        if: ${{ github.ref == 'refs/heads/main' }}
         run: >-
           aws s3 sync
           ${BIN_DIR} s3://${{ secrets.PREPROD_AWS_S3_BUCKET }}/
           --no-progress
 
       - name: Invalidate downloads.spacelift.dev cache
+        if: ${{ github.ref == 'refs/heads/main' }}
         run: >-
           aws cloudfront create-invalidation
           --distribution-id ${{ secrets.PREPROD_DISTRIBUTION }}
           --paths "/*"
 
+      - name: Log in to Amazon public ECR
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+
+    #  - name: Set up QEMU
+    #    uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and push the image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          platforms: linux/amd64,linux/arm64
```

Comment:
Do we really want to build for `arm64` here? I think we're currently only building an `amd64` binary, so I don't think the ARM docker image will work correctly.